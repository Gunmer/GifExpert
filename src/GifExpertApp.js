import {useState} from "react";
import AddCategory from "./componets/AddCategory";
import {GifGrid} from "./componets/GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['star wars']);

  return (
      <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories} />
        <hr/>

          {
            categories.map(category => <GifGrid key={category} category={category} />)
          }
      </>
  );
};

export default GifExpertApp
