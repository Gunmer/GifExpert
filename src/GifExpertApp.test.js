import GifExpertApp from "./GifExpertApp";
import {shallow} from "enzyme";

describe('GifExpertApp', () => {
  it('should be render', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot()
  });

  it('should show a category list', () => {
    const categories = ['Star Wars'];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper.find('GifGrid')).toHaveLength(categories.length);
    expect(wrapper).toMatchSnapshot();
  });
});
