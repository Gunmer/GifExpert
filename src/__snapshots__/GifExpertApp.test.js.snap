// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should be render 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`GifExpertApp should show a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Star Wars"
    key="Star Wars"
  />
</Fragment>
`;
